### ISSUE TITLE
Macro "Cuts" in Macro Carts. Possible feature request.

### ISSUE THREAD

**Issue by zotz** [ElvishArtisan/rivendell#538]
Macro "Cuts" in Macro Carts.

Currently macro carts have cart numbers but no "cut numbers" ...

would it be possible to have macro cuts in the macro carts where the macro lines go now and then have the current macro lines go in the macro cuts?

This would let us create "rotators" in macro carts. And to other cut related stuff like dayparting, etc. the same as we do for audio cuts now...

all the best,

drew

**Comment by ElvishArtisan** [comment]
If you need complex dayparting, a better approach would be to implement it in a script and call it with ```RN```.

**Comment by zotz** [comment]
"If you need complex dayparting, a better approach would be to implement it in a script and call it with RN."

the dayparting I had in mind was the same as is available for audio carts now... Basically, giving macro carts the same features that audio carts get.

So, multiple cuts.

Cuts have Air Date/Time

Daypart

Day of the Week

Options... Is that what you meant when mentioning complex dayparting?

Just want us to be on the same page before proceeding...

**Comment by ElvishArtisan** [comment]
> the dayparting I had in mind was the same as is available for audio carts now... Basically, giving macro carts the same features that audio carts get.

Yes, that was my understanding. My basic design rationale here (and indeed, through all of Rivendell) is to makes things "as simple as possible, but no simpler". For audio play-outs, that means built-in cut dayparting, because there really is no other, simpler way to provide that kind of functionality efficiently. Macro carts are different though. Being effectively *programs*, it's (relatively) easy to build dayparting into the scripts themselves. Thus, we kick the can of complexity down the road to those scripts that actually *need* such functionality (a small minority; most sites in my experience never need this functionality at all), rather than making the whole system bear the cost. This also lets such scripts implement the specific *kind* of dayparting they need for the specific task at hand, rather then constraining them into a 'one size fits all' model imposed by the automation system.

These design trade-offs are admittedly matters of engineering judgement much more than cut and dried prescription. For a thoughtful discussion of them from a point of view firmly within the UNIX design tradition, see [Chapter 13](http://www.catb.org/esr/writings/taoup/html/complexitychapter.html) Eric Raymond's [The Art of UNIX Programming](http://www.catb.org/esr/writings/taoup/html/).
